src/handlers/del.ts

```diff
@@ -0,0 +1,37 @@
+import path from 'node:path';
+import { unlink } from 'node:fs/promises';
+import { defineHandler, storePath } from './common.ts';
+
+export default defineHandler({
+  name: 'del',
+
+  test(requestUrl, request) {
+    return request.method === 'POST' && requestUrl.pathname === '/delete';
+  },
+
+  async handle(requestUrl, request) {
+    const body = await request.json();
+
+    const urlsArray = body.urls;
+
+    if (urlsArray.length) {
+      for (let url of urlsArray) {
+        const formattedUrl = url.replace(requestUrl.origin + '/', '');
```

Comment:
```ts
const urlObject = new URL(url);
const pathname = urlObject.pathname; // your formattedUrl with a prefix '/'
// Optional: verify the origin
// assert urlObject.origin === requestUrl.origin
```

Reply:
Del function from Vercel blob can take as an argument whole url or pathname. If you try to make a URL object form pathname you'll get an error. 
But anyway, I've refactored code a little, seems more readable now.